parse_contributors_oo.py

```diff
@@ -0,0 +1,313 @@
+import json
+import ruamel.yaml
+import urllib.request
+import requests
+
+
+class processContributors:
+    # When initializing how do you decide what should be an input
+    # attribute vs just something a method accepted when called?
+    def __init__(self, json_files: list, API_TOKEN: str):
+        self.json_files = json_files
+        self.API_TOKEN = API_TOKEN
+
+    def process_json_file(self, json_file: str) -> dict:
+        """Deserialize a JSON file from a URL and cleanup data
+
+        Open a JSON file containing contributor data created from a all-contributors
+        json file. Rename fields to match fields used in the website. Then
+        add keys needed for the website.
+
+        # TODO: if you use types do you need a docstring still with params?
+        Parameters
+        ----------
+        json_file : string
+            Web url of a json formatted file
+
+        """
+        response = requests.get(json_file)
+        data = json.loads(response.text)
+
+        # Create a dictionary to hold the processed data
+        processed_data = {}
+        # Loop through each entry in the JSON file
+        for entry in data["contributors"]:
+            # Check if the login value is already in the dictionary
+            if entry["login"] not in processed_data:
+                # Rename the login key to github_username
+                entry["github_username"] = entry.pop("login")
+
+                # Rename the profile key to website
+                entry["website"] = entry.pop("profile")
+
+                # Process github image avatar id
+                entry["avatar_url"] = int(
+                    entry["avatar_url"].rsplit("/", 1)[-1].rsplit("?", 1)[0]
+                )
+                entry["github_image_id"] = entry.pop("avatar_url")
+                # Add empty values for the new keys
+                for key in [
+                    "mastodon",
```

Comment:
keys probably shouldn't be hard coded into this method. it is also long because of that